app/src/main/java/com/ddoczi/tasky/authentication/presentation/login/LoginViewModel.kt

```diff
@@ -60,6 +66,20 @@ class LoginViewModel @Inject constructor(
     }
 
     private fun submit(email: String, password: String) {
-        // TODO
+        viewModelScope.launch {
+            repository.login(email, password)
+                .onSuccess {
+                    println(it)
+                    preferences.saveToken(it.token)
+                    preferences.saveFullName(it.fullName)
+                    preferences.saveUserId(it.userId)
```

Comment:
Could it make sense to expose a single save function here that takes in a data class for all these single fields? I don't think you'll ever want to change just one field in this app. You either want to change them all together or none of them.
I personally did something like this:
```
suspend fun saveUserInfo(user: AuthenticatedUser)
```
With this wrapper:
```
data class AuthenticatedUser(
    val token: String,
    val fullName: String,
    val email: String
)
```